docs/esi-leap-api-ref.md

```diff
@@ -10,58 +10,82 @@ The Offer API endpoint can be reached at /v1/offers.
   * resource_uuid: the uuid of the resource. If using with Ironic this would be the node's uuid.
     * A string.
     * This field is required.
+  * resource_type:
+    * A string.
+    * This field is optional. Not setting it will default to 'ironic_node'.
+  * name:
+    * A string.
+    * This field is optional.
+  * lessee_id:
+    * A string.
+    * This field is optional.
+  * resource_class:
+    * A string.
+    * This field is optional.
+  * resource_properties:
+    * A JSON object.
+    * This field is optional.
   * start_time:
     * A datetime string.
     * This field is optional. Not setting it will default start_time to the time when the request was sent.
   * end_time:
     * A datetime string.
     * This field is optional. Not setting it will default end_time to datetime.max.
   * properties:
-    * a json object
+    * A JSON object.
     * This field is optional. Not setting it will default to {}.
-* 'status', 'uuid', and 'availabilities' are read only.
-* The response to a POST request will be the newly created offer. The response type is 'application/json'.
-
-An example curl request is shown below.
-```
-curl -X POST -sH "X-Auth-Token: $token" http://localhost:7777/v1/offers  -H 'Content-Type: application/json' -d '{
-  "resource_type": "dummy_node",
-  "resource_uuid": "1718",
-  "name": "o1",
-  "start_time": "2020-04-07 05:45:02",
-  "end_time": "2020-04-09 05:45:02",
-   "properties": {
-      "memory_gb": 10240,
-      "cpu_arch": "x86_64",
-      "cpu_physical_count": 4,
-      "cpu_core_count": 16,
-      "cpu_ghz": 3,
-      "disks": [
+  * 'status', 'uuid', 'availabilities', 'project_id', 'project', 'lessee', resource', and 'parent_lease_uuid' are read only.
+  * The response to a POST request will be the newly created offer. The response type is 'application/json'.
+
+  * An example curl request is shown below.
+    ```
+    curl -X POST -sH "X-Auth-Token: $token" http://localhost:7777/v1/offers  -H 'Content-Type: application/json' -d '{
+      "resource_type": "dummy_node",
+      "resource_uuid": "1718",
+      "name": "o1",
+      "start_time": "2020-04-07 05:45:02",
+      "end_time": "2020-04-09 05:45:02",
+      "properties": {
+          "memory_gb": 10240,
+          "cpu_arch": "x86_64",
+          "cpu_physical_count": 4,
+          "cpu_core_count": 16,
+          "cpu_ghz": 3,
+          "disks": [
+            {
+              "size_gb": 500,
+              "model": "YOYODYNE 1234"
+            },
         {
-          "size_gb": 500,
-          "model": "YOYODYNE 1234"
-        },
-    {
-          "size_gb": 1024,
-          "model": "evo840 ssd"
+              "size_gb": 1024,
+              "model": "evo840 ssd"
+            }
+          ]
         }
-      ]
-    }
-}'
-```
+    }'
+    ```
+* The /v1/offers/\<uuid>/claim endpoint supports POST requests to claim the offer with the given uuid. The body of the request contains the lease information with the given values.
```

Comment:
I would format this section so there's an actual header, so it's more apparent that there's another API function here. Right now it's a bit lost within the `POST` section. You can take a look at https://docs.openstack.org/api-ref/baremetal/ for an example of how specialized API commands are listed.

Reply:
I added headers to each endpoint to make it structured in a similar way to the Baremetal API documentation. Let me know if this fine or too much.